data/publisher/src/test/java/tech/pegasys/teku/data/publisher/MetricsDataFactoryTest.java

```diff
@@ -18,51 +18,50 @@
+  //  private final JsonProvider jsonProvider = new JsonProvider();
   PrometheusMetricsSystem prometheusMock = mock(PrometheusMetricsSystem.class);
 
   @Test
   public void shouldExtractMetricsFromPrometheusMetricsAndSerialiseJson()
       throws JsonProcessingException {
     when(prometheusMock.streamObservations()).thenReturn(getMockObservations().stream());
     final MetricsDataFactory metricsDataFactory = new MetricsDataFactory(prometheusMock);
-    final BaseMetricData baseMetricData =
-        metricsDataFactory.getMetricData(MetricsDataClient.VALIDATOR);
-    final String data = jsonProvider.objectToJSON(baseMetricData);
-    assertThat(baseMetricData)
-        .isEqualTo(jsonProvider.jsonToObject(data, ValidatorMetricData.class));
-    assertThat(baseMetricData).isInstanceOf(ValidatorMetricData.class);
-    ValidatorMetricData validatorMetricData = (ValidatorMetricData) baseMetricData;
-    assertThat(validatorMetricData.cpu_process_seconds_total).isEqualTo(1);
-    assertThat(validatorMetricData.validator_total).isEqualTo(1);
-    assertThat(validatorMetricData.validator_active).isEqualTo(1);
+    final List<BaseMetricData> baseMetricData = metricsDataFactory.getMetricData();
+
+    ObjectMapper mapper = new ObjectMapper();
+    final String data = mapper.writeValueAsString(baseMetricData);
+
+    assertThat(data).contains("beaconnode");
+    assertThat(data).contains("validator");
+    assertThat(data).contains("\"validator_active\":1");
+    assertThat(data).contains("\"disk_beaconchain_bytes_total\":0");
   }
 
   @Test
   public void shouldSerializeObjectFromPrometheusMetricsWithDefaultValues()
       throws JsonProcessingException {
     when(prometheusMock.streamObservations()).thenReturn(new ArrayList<Observation>().stream());
     final MetricsDataFactory metricsDataFactory = new MetricsDataFactory(prometheusMock);
-    final BaseMetricData baseMetricData =
-        metricsDataFactory.getMetricData(MetricsDataClient.VALIDATOR);
-    final String data = jsonProvider.objectToJSON(baseMetricData);
-    assertThat(baseMetricData)
-        .isEqualTo(jsonProvider.jsonToObject(data, ValidatorMetricData.class));
-    assertThat(baseMetricData).isInstanceOf(ValidatorMetricData.class);
-    ValidatorMetricData validatorMetricData = (ValidatorMetricData) baseMetricData;
-    assertThat(validatorMetricData.cpu_process_seconds_total).isEqualTo(0);
-    assertThat(validatorMetricData.validator_total).isEqualTo(0);
-    assertThat(validatorMetricData.validator_active).isEqualTo(0);
+    final List<BaseMetricData> baseMetricData = metricsDataFactory.getMetricData();
+
+    ObjectMapper mapper = new ObjectMapper();
+    final String data = mapper.writeValueAsString(baseMetricData);
+
+    assertThat(data).contains("beaconnode");
+    assertThat(data).contains("validator");
+    assertThat(data).contains("\"validator_active\":0");
+    assertThat(data).contains("\"disk_beaconchain_bytes_total\":0");
```

Comment:
We should deserialise the JSON and then assert on the values rather than trying to do string matches into the JSON (which may fail due to formatting differences).